cmd/cloudshell_open/artifactregistry.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+
+	artifactregistry "cloud.google.com/go/artifactregistry/apiv1"
+	artifactregistrypb "cloud.google.com/go/artifactregistry/apiv1/artifactregistrypb"
+)
+
+// Create a "Cloud Run Source Deploy" repository in Artifact Registry (if it doesn't already exist)
+func createArtifactRegistry(project string, region string, repoName string) error {
+
+	repoPrefix := fmt.Sprintf("projects/%s/locations/%s", project, region)
+	repoFull := fmt.Sprintf("%s/%s", repoPrefix, repoName)
+
+	ctx := context.Background()
+
+	//Check for existing repo
+	client, err := artifactregistry.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to create artifact registry client: %w", err)
+	}
+
+	req := &artifactregistrypb.GetRepositoryRequest{
+		Name: repoFull,
+	}
+	var existingRepo, _ = client.GetRepository(ctx, req)
```

Comment:
Nit: a short variable declaration here would be idiomatic and more consistent with the other declarations in this function.